test/helpers.js

```diff
@@ -0,0 +1,254 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+assert.throwsAsync = (fn, errorLike, errMsgMatcher, message) => {
+  let threw = null;
+  return fn().catch(err => {
+    threw = err;
+  }).then(() => {
+    // Use synchronous `assert.throws` to get all the nice matching logic.
+    assert.throws(() => {
+      if (threw) {
+        throw threw;
+      }
+    }, errorLike, errMsgMatcher, message);
+  });
+};
+
+// These are helper functions for generating a variety of byte structs
+// used by the protocol, optionally with a variety of errors or irregularities
+// for testing purposes.
+//
+// The code here duplicates some of the logic of the classes we use for
+// creating such structs in the source codebase itself, and a future refactor
+// might enable us to reduce that duplication.  For now though, it seems better
+// to suffer such duplication than to add things to the source codebase that
+// might accidentally generate badly-formed data in production use.
+
+const testHelpers = {
+
+  tamper: function (bytes, start = 0) {
+    // Make one random change to a byte string.
+    // Randomness is good when we want *any* change to cause a failure.
+    const tampered = bytes.slice();
+    const i = Math.floor(Math.random() * (tampered.byteLength - start) + start);
+    tampered[i] += 1 + Math.floor(Math.random() * 255);
+    return tampered;
```

Comment:
The problem with this function is that if it does trigger an error, then you won't ever know what conditions produced that error.  If you are going to fuzz, record the way in which your tampering was applied so that you can reproduce problems easily.